tensorzero-core/src/experimentation/track_and_stop/check_stopping.rs

```diff
@@ -0,0 +1,496 @@
+#![allow(dead_code)]
+
+use thiserror::Error;
+
+use super::estimate_optimal_probabilities::argmax;
+use crate::db::FeedbackByVariant;
+
+pub struct PairwiseGLRArgs {
+    leader_pulls: u64,
+    leader_mean: f64,
+    leader_variance: f64,
+    challenger_pulls: u64,
+    challenger_mean: f64,
+    challenger_variance: f64,
+    epsilon: f64,
+}
+
+#[derive(Debug, Error)]
+pub enum PairwiseGLRError {
+    #[error("Error in compute_pairwise+_glr")]
+    GLRError,
+}
+
+#[derive(Debug, Error)]
+pub enum CheckStoppingError {
+    #[error("Error in computing pairwise GLR")]
+    GLRError,
+    #[error("Error in checking stopping condition")]
+    StoppingError,
+}
+
+fn compute_pairwise_glr(args: PairwiseGLRArgs) -> Result<f64, PairwiseGLRError> {
+    let PairwiseGLRArgs {
+        leader_pulls,
+        leader_mean,
+        leader_variance,
+        challenger_pulls,
+        challenger_mean,
+        challenger_variance,
+        epsilon,
+    } = args;
+    // TODO: right way to validate inputs here?
+    if leader_mean < challenger_mean {
+        return Err(PairwiseGLRError::GLRError);
+    }
+
+    // No data means no evidence
+    if (leader_pulls == 0) | (challenger_pulls == 0) {
+        return Ok(0.0);
+    }
+
+    // Compute the empirical gap (with ε-adjustment)
+    let empirical_gap: f64 = leader_mean - challenger_mean + epsilon;
+
+    // Only consider positive gaps (evidence for H₁)
+    if empirical_gap <= 0.0 {
+        return Ok(0.0);
+    }
+
+    // Compute pooled standard error
+    let pooled_variance: f64 = leader_variance / (leader_pulls.max(1) as f64)
+        + challenger_variance / (challenger_pulls.max(1) as f64);
+
+    if pooled_variance <= 0.0 {
+        return Ok(0.0);
+    }
+
+    // GLR statistic: (gap)^2 / (2 * pooled_variance)
+    let glr_statistic: f64 = (empirical_gap.powi(2)) / (2.0 * pooled_variance);
+
+    Ok(glr_statistic.max(0.0))
+}
+
+pub struct StoppingResult {
+    can_stop: bool,
+    leader_arm: usize,
+}
+
+///     Decide whether to stop the experiment and which arm to recommend.
+///     Uses parallel GLR testing with uniform challenger weights π_j = 1/(K-1),
+///     and per-pair time t_{L,j} = n_L + n_j.
+pub fn check_stopping(
+    feedback: Vec<FeedbackByVariant>,
+    min_pulls: u64,
+    ridge_variance: Option<f64>,
+    epsilon: Option<f64>,
+    delta: Option<f64>,
+) -> Result<StoppingResult, CheckStoppingError> {
```

Comment:
do you mind refactoring to have the function return a `Result<StoppingResult, CheckStoppingError>` where 
```
pub enum StoppingResult {
    NotStopped,
    Winner(String),
}
```